CustomDeserializers/AvroExample/EventhubCaptureReader.cs

```diff
@@ -0,0 +1,61 @@
+﻿//********************************************************* 
+// 
+//    Copyright (c) Microsoft. All rights reserved. 
+//    This code is licensed under the Microsoft Public License. 
+//    THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF 
+//    ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY 
+//    IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR 
+//    PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT. 
+// 
+//********************************************************* 
+
+
+using System.Collections.Generic;
+using System.IO;
+
+using Avro.Generic;
+using Avro.IO;
+using Avro.File;
+using Microsoft.Azure.StreamAnalytics;
+using Microsoft.Azure.StreamAnalytics.Serialization;
+
+namespace ExampleCustomCode.AvroSerialization
+{
+    // Reads eventhub capture format. https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-capture-overview
+    public abstract class EventhubCaptureReader<T> : StreamDeserializer<T>
+    {
+        public override IEnumerable<T> Deserialize(Stream stream)
+        {
+            var reader = DataFileReader<GenericRecord>.OpenReader(stream);
+            foreach(GenericRecord genericRecord in reader.NextEntries)
+            {
+                var eventData = new EventData()
+                {
+                    EnqueuedTimeUtc = (string)genericRecord["EnqueuedTimeUtc"],
+                    Offset = (string)genericRecord["Offset"],
+                    SequenceNumber = (long)genericRecord["SequenceNumber"],
```

Comment:
error handling?

Reply:
In this case, there isn't anything we can do when there is an error.
Eventhub capture is expected to be in this format, if it is not and there is failure, it is better to fail and let the exception propagate.

Was there anything specific that you wanted handled?